src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -3141,6 +3143,9 @@ flush_result ntp_archiver::flush() {
           .response = flush_response::rejected, .offset = std::nullopt};
     }
 
+    _local_segment_merger->set_enabled(false);
+    _scrubber->set_enabled(false);
```

Comment:
should these be re-enabled if for some reasons the migration was not completed after flush?

Reply:
it will be re-enabled when the new term will start